server/controllers/userFollow.js

```diff
@@ -0,0 +1,168 @@
+import isEmpty from "lodash.isempty";
+import models from "@models";
+import { serverError, httpResponse } from "@helpers/http";
+
+const { Followers, Users } = models;
+
+/**
+ * @description Follow a user
+ *
+ * @param {object} req HTTP request object
+ * @param {object} res HTTP response object
+ * @returns {object} HTTP response
+ */
+export const followUser = async (req, res) => {
+  const { userId } = req.user;
+  const authorId = req.userDetails.id;
+  try {
+    if (userId === authorId) {
+      return httpResponse(res, {
+        statusCode: 403,
+        success: false,
+        message: "You cannot follow yourself"
+      });
+    }
+    const follower = await Followers.findOne({
+      where: { authorId, followerId: userId }
+    });
+    if (isEmpty(follower)) {
+      await Followers.create({
+        authorId,
+        followerId: userId
+      });
+      return httpResponse(res, {
+        statusCode: 201,
+        success: true,
+        message: "You are now following the author"
+      });
+    }
+    return httpResponse(res, {
+      statusCode: 404,
+      success: false,
+      message: "Sorry, you are following this author"
```

Comment:
` "Sorry, you are following this author"` can be changed to ` "Sorry, you are already following this author"` to make it clear that the action is a **_repetition_** of what has been done already.

Reply:
It has been implemented